### PR TITLE
Only render DateTimeWidget value with milliseconds for fractional types

### PR DESCRIPTION
**Pull request by othercorey**
https://github.com/cakephp/cakephp/issues/14226

### CODE REVIEW

File: src/View/Widget/DateTimeWidget.php [hunk 1]
```diff
@@ -192,7 +194,21 @@ protected function formatDateTime($value, array $options): string
             $dateTime = $dateTime->setTimezone($timezone);
         }
 
-        return $dateTime->format($this->formatMap[$options['type']]);
+        $format = $this->formatMap[$options['type']];
+        if ($fieldName !== null) {
+            $fractionalTypes = [
```
Comment (ADmad) [inline src/View/Widget/DateTimeWidget.php]: This array is now duplicated at 2 places in the class, so better to set it to a property. If the same set it used at other places in the framework too?

Comment (othercorey) [reply src/View/Widget/DateTimeWidget.php]: I thought this too, but wasn't sure if it was good style to make a list of fractional datetime types in the schema.

I do want something where we don't miss new types again though.

Comment (markstory) [reply src/View/Widget/DateTimeWidget.php]: Could TableSchema have a constant that contains the fractional types?

**Comment by ADmad** [comment]
I was wondering if we could take a simpler approach to fix the issue. Simply truncate the milliseconds from the formatted values if values of `step` is an integer / not decimal. That would be just a 2-3 lines patch.

**Comment by othercorey** [comment]
> Simply truncate the milliseconds from the formatted datetime string if value of `step` attribute is an integer / not decimal. That would be just a 2-3 lines patch.

If users set the step to 0.5 but had a datetime type, the insert/update would fail.

**Comment by ADmad** [comment]
> If users set the step to 0.5 but had a datetime type, the insert/update would fail.

That's users fault for setting invalid attribute.

**Pull request merged**